docs/kernel_providers.rst

```diff
@@ -0,0 +1,149 @@
+================
+Kernel providers
+================
+
+.. note::
+   This is a new interface under development. Not all Jupyter applications
+   use this yet. See :ref:`kernelspecs` for the established way of discovering
+   kernel types.
+
+By writing a kernel provider, you can extend how Jupyter applications discover
+and start kernels. To do so, subclass
```

Comment:
Perhaps add one short sentence of why one might wish to create a kernel provider.

Add a subheading: Create a kernel provider